backend/modules/query_controllers/multimodal/controller.py

```diff
@@ -180,38 +201,31 @@ async def _get_retriever(self, vector_store, retriever_name, retriever_config):
 
         return retriever
 
-    async def _stream_answer(self, rag_chain, query):
-        async with async_timeout.timeout(GENERATION_TIMEOUT_SEC):
-            try:
-                async for chunk in rag_chain.astream(query):
-                    if "question " in chunk:
-                        # print("Question: ", chunk['question'])
-                        yield json.dumps({"question": chunk["question"]})
-                    elif "context" in chunk:
-                        # print("Context: ", self._format_docs_for_stream(chunk['context']))
-                        yield json.dumps(
-                            {"docs": self._format_docs_for_stream(chunk["context"])}
-                        )
-                    elif "answer" in chunk:
-                        # print("Answer: ", chunk['answer'])
-                        yield json.dumps({"answer": chunk["answer"]})
-
-                yield json.dumps({"end": "<END>"})
-            except asyncio.TimeoutError:
-                raise HTTPException(status_code=504, detail="Stream timed out")
+    def _cleanup_metadata(self, docs):
+        formatted_docs = list()
+        for doc in docs:
+            metadata = {
+                key: doc.metadata[key]
+                for key in self.required_metadata
+                if key in doc.metadata
+            }
+            formatted_docs.append(
+                Document(page_content=doc.page_content, metadata=metadata)
+            )
+        return formatted_docs
+
+    async def _sse_wrap(self, gen):
+        async for data in gen:
+            yield "event: data\n"
+            yield f"data: {json.dumps(data.dict())}\n\n"
+        yield "event: end\n"
```

Comment:
Let's make this part of Base?

Reply:
We do not have a Base class yet, need to this as refactor to push common functionalities to Base.